### PR TITLE
Use load in mo logger 

### PR DESCRIPTION
**Pull request by gavinyue**
## What type of PR is this?

- [ ] API-change
- [x] BUG
- [x] Improvement
- [ ] Documentation
- [ ] Feature
- [ ] Test and CI
- [ ] Code Refactoring

## Which issue(s) this PR fixes:

issue #https://github.com/matrixorigin/matrixone/issues/11628

## What this PR does / why we need it:

Currently, using prepare and commit will open long-running tx in CN side, which cause some problem for CN.  So migrate to load to simplify the logic here.

### CODE REVIEW

File: pkg/util/export/etl/db/db_holder.go [hunk 1]
```diff
@@ -254,162 +254,87 @@ func getPrepareSQL(tbl *table.Table, columns int, batchLen int, middleBatchLen i
...
+
+	csvWriter := NewCSVWriter(ctx)
+	defer csvWriter.Release() // Ensures that the buffer is returned to the pool
+
+	// Write each record to the CSVWriter
+	for _, record := range records {
+		if err := csvWriter.WriteStrings(record); err != nil {
+			return err
 		}
 	}
```
Comment (fengttt) [inline pkg/util/export/etl/db/db_holder.go:328]: Here, can you add one more line to the csvData, basically that line should be equivalent as if  it is
log.Infof("Log wrote %d records %d bytes", x, y)

File: pkg/util/export/etl/db/db_holder.go [hunk 2]
```diff
@@ -330,11 +330,25 @@ func bulkInsert(ctx context.Context, sqlDb *sql.DB, records [][]string, tbl *tab
 	escapedCSVData := strings.ReplaceAll(csvData, "'", "''")
 	loadSQL := fmt.Sprintf("LOAD DATA INLINE FORMAT='csv', DATA='%s' INTO TABLE %s.%s", escapedCSVData, tbl.Database, tbl.Table)
 
-	// Execute the LOAD DATA command directly without a transaction
-	_, err := sqlDb.Exec(loadSQL)
+	// Begin a new transaction
```
Comment (fengttt) [inline pkg/util/export/etl/db/db_holder.go:333]: why?   Isn't this a single command, so auto commit mode works just fine?

**Review by fengttt — APPROVED** [review]

File: pkg/util/export/etl/db/db_holder.go (hunk 2 above)
Comment (gavinyue) [reply pkg/util/export/etl/db/db_holder.go:376]: There are duplicates. I do not know where they are from, so added the tx here.  Still unknown...

**Comment by xzxiong** [comment]
@gavinyue pls add some basic background desc.

looks like replacing `insert into ...` with `LOAD DATA INLINE FORMAT='csv', DATA='%s' INTO TABLE %s.%s`.

File: pkg/util/export/etl/db/db_holder.go (hunk 2 above)
Comment (fengttt) [reply pkg/util/export/etl/db/db_holder.go]: duplicate?   what is the error message?   the table does not have any primary key etc so where does duplicates come from?

File: pkg/util/export/etl/db/db_holder.go [hunk 3]
```diff
@@ -181,235 +150,131 @@ func WriteRowRecords(records [][]string, tbl *table.Table, timeout time.Duration
 	ctx, cancel := context.WithTimeout(context.Background(), timeout)
 	defer cancel()
 
-	err = bulkInsert(ctx, dbConn, records, tbl, MaxInsertLen, MiddleInsertLen)
+	err = bulkInsert(ctx, dbConn, records, tbl)
 	if err != nil {
+		fmt.Printf("gavinyue bulkInsert failed, err: %v\n", err)
```
Comment (nnsgmsone) [inline pkg/util/export/etl/db/db_holder.go:155]: It seems that the use of fmt.Printf to print information is not allowed

Comment (gavinyue) [reply pkg/util/export/etl/db/db_holder.go]: removed

File: pkg/util/export/etl/db/db_holder.go [hunk 4]
```diff
@@ -190,226 +159,121 @@ func WriteRowRecords(records [][]string, tbl *table.Table, timeout time.Duration
...
-			sqls = getPrepareSQL(tbl, len(records[0]), batchLen, middleBatchLen)
-			prepareSQLMap.Store(key, sqls)
+
+	csvWriter := NewCSVWriter(ctx)
+	defer csvWriter.Release() // Ensures that the buffer is returned to the pool
+
+	// Write each record of the chunk to the CSVWriter
+	for _, record := range records {
+		for i, col := range record {
+			record[i] = strings.ReplaceAll(strings.ReplaceAll(col, "\\", "\\\\"), "'", "''")
```
Comment (xzxiong) [inline pkg/util/export/etl/db/db_holder.go:230]: 是否可以用 prepare 语句，解决这里字符转义的问题？
感觉这里替换起来也是没完没了。

File: pkg/util/export/etl/db/db_holder.go [hunk 5]
```diff
@@ -190,226 +159,121 @@ func WriteRowRecords(records [][]string, tbl *table.Table, timeout time.Duration
...
-			}
-			err = oneStmt.Close()
-			if err != nil {
-				tx.Rollback()
-				return err
+	retryCount := 2
+	var lastError error
+
+	for i := 0; i < retryCount; i++ {
+		txn, err := sqlDb.Begin()
```
Comment (xzxiong) [inline pkg/util/export/etl/db/db_holder.go:247]: 为什么要坚持使用 begin-commit？Load语句本身是事务性的。

**Review by xzxiong — APPROVED** [review]

File: pkg/util/export/etl/db/db_holder.go (hunk 2 above)
Comment (gavinyue) [reply pkg/util/export/etl/db/db_holder.go]: It does help. And retry helps even more.

File: pkg/util/export/etl/db/db_holder.go (hunk 5 above)
Comment (gavinyue) [reply pkg/util/export/etl/db/db_holder.go:247]: 不用的话 就是有时候会有duplicate.

File: pkg/util/export/etl/db/db_holder.go (hunk 2 above)
Comment (gavinyue) [reply pkg/util/export/etl/db/db_holder.go]: But after fixing escape issue, I do not see the duplicates anymore.  So may try to remove it to test

File: pkg/util/export/etl/db/db_holder.go (hunk 4 above)
Comment (gavinyue) [reply pkg/util/export/etl/db/db_holder.go:230]: 没看到prepare支持, 我尝试了半天也不work.   MYSQL的prepare也不支持load data.

File: pkg/util/export/etl/db/db_holder.go (hunk 2 above)
Comment (gavinyue) [reply pkg/util/export/etl/db/db_holder.go]: Should be no problem now

**Review by nnsgmsone — APPROVED** [review]

File: pkg/util/export/etl/db/db_holder.go (hunk 1 above)
Comment (gavinyue) [reply pkg/util/export/etl/db/db_holder.go:236]: resolved as discussed

**Pull request merged**